### PR TITLE
returned the results of `asyncio.wait` in `bps.wait_for`

### PR DESCRIPTION
**Pull request by evalott100**
Also updated docs to reflect this, and added a test.

### CODE REVIEW

File: src/bluesky/tests/test_run_engine.py [hunk 1]
```diff
@@ -1956,6 +1957,41 @@ def plan():
...
+    timeout_task = asyncio.Task(timeout_future())
+
+    def runtime_error_and_success_plan():
+        statuses = yield from wait_for([lambda: passing_future(), lambda: failing_future()], timeout=0.1)
+        assert isinstance(statuses, set) and len(statuses) == 2
+        passing_future_status = [x for x in statuses if x.get_coro().__name__ == "passing_future"][0]
+        failing_future_status = [x for x in statuses if x.get_coro().__name__ == "failing_future"][0]
+        assert passing_future_status.done() and not passing_future_status.exception()
+        assert passing_future_status.result() == 1080
+        assert failing_future_status.done() and failing_future_status.exception()
```
Comment (coretl) [inline src/bluesky/tests/test_run_engine.py:1980]: Can we check that task references passed through are maintained please?

```suggestion
        passing_task = asyncio.create_task(passing_future())
        failing_task = asyncio.create_task(failing_future())
        tasks = yield from wait_for([lambda: passing_task, lambda: failing_task], timeout=0.1)
        assert tasks == {passing_task, failing_task}
        assert passing_task.done() and not passing_task.exception()
        assert passing_task.result() == 1080
        assert failing_task.done() and failing_task.exception()
```

**Review by coretl — COMMENTED** [review]
Looks good! I suggested a slight clarification to the test

**Comment by coretl** [comment]
https://github.com/numpy/numpydoc/pull/542 means yields and returns should now be supported in docstring. Is numpydoc released with that pr in?

**Comment by evalott100** [comment]
> [numpy/numpydoc#542](https://github.com/numpy/numpydoc/pull/542) means yields and returns should now be supported in docstring. Is numpydoc released with that pr in?

Nah, that PR is scheduled for release in 1.8.0.

File: src/bluesky/run_engine.py [hunk 2]
```diff
@@ -1798,9 +1798,10 @@ def fut_fac() -> awaitable:
 
         (futs,) = msg.args
         futs = [asyncio.ensure_future(f()) for f in futs]
-        _, pending = await asyncio.wait(futs, **self._loop_for_kwargs, **msg.kwargs)
+        completed, pending = await asyncio.wait(futs, **self._loop_for_kwargs, **msg.kwargs)
         if pending:
             raise WaitForTimeoutError("Plan failed to complete in the specified time")
+        return completed
```
Comment (coretl) [inline src/bluesky/run_engine.py:1804]:
```suggestion
        return futs
```
to keep ordering?

**Review by coretl — COMMENTED** [review]
And point numpydoc to main in dev dependencies so you can use Yields: and Returns:

File: src/bluesky/tests/test_run_engine.py [hunk 3]
```diff
@@ -1956,6 +1957,45 @@ def plan():
...
+        async def failing_coro():
+            await asyncio.sleep(0.001)
+            raise RuntimeError
+
+        failing_task = asyncio.create_task(failing_coro())
+
+        tasks = yield from wait_for([lambda: passing_task, lambda: failing_task])
+        assert passing_task is tasks[0] and failing_task is tasks[1]
+        assert passing_task.done() and passing_task.result() == 1080
+        assert failing_task.done() and isinstance(failing_task.exception(), RuntimeError)
```
Comment (coretl) [inline src/bluesky/tests/test_run_engine.py:1977]: Now you've made the returning futures change, please can we check that passing coros and returning classes works:
```suggestion
    def runtime_error_and_success_plan():
        async def passing_coro():
            await asyncio.sleep(0.001)
            return 1080

        async def failing_coro():
            await asyncio.sleep(0.001)
            raise RuntimeError

        passing_task, failing_task = yield from wait_for([lambda: passing_coro(), lambda: failing_coro()])
        assert passing_task.get_coro().__name__ == "passing_coro"
        assert passing_task.done() and passing_task.result() == 1080
        assert failing_task.get_coro().__name__ == "failing_coro"
        assert failing_task.done() and isinstance(failing_task.exception(), RuntimeError)
```

**Comment by coretl** [comment]
> The newer numpydoc requires `python>=3.9`, which is why the 3.8 build is failing here.
> 
> I don't know what the bluesky deprecation schedule for python versions is like - time to take out 3.8?
> 
> @danielballan @tacaswell @coretl

Decided to stick with 3.8 until it end of lifes in October, so will change the docstring back

**Review by coretl — APPROVED** [review]

**Pull request merged**